.github/workflows/unit_tests.yml

```diff
@@ -2,11 +2,7 @@ name: Unit tests
 
 on:
   push:
-    branches:
-      - master
```

Comment:
This I strongly prefer to keep. I don't see any reason to test both on pushes and pull requests to any branch. What I think we should do instead is to add an additional line both here and below:
```yaml
on:
  push:
    branches:
      - master
      - releases/*
```
People wanting to test a local branch before pushing can use `tox` directly.

Reply:
Well, no, since that requires that i) they have `tox` installed (and maybe other stuff), ii) that they have several Python versions.

I find it very easy that the tests run in GitHub Actions, just like they are for `master`. And it doesn't stop me from running the tests locally.

What's the downside of always testing, why restrict this to just `master` and releases? It's not that the tests take forever...

Reply:
> Well, no, since that requires that i) they have tox installed (and maybe other stuff), ii) that they have several Python versions.

Not a huge request imho if you develop a Python package.

> I find it very easy that the tests run in GitHub Actions, just like they are for master. And it doesn't stop me from running the tests locally.

I think it's wasteful, and besides that every branch will be in the end PRd to either `master` or `releases` and will be tested at that point. Can we at least move this single change to another PR and eventually continue the discussion there (so that the other changes here can be merged)?

Reply:
> Can we at least move this single change to another PR and eventually continue the discussion there (so that the other changes here can be merged)?

@boegel :point_up:

Reply:
I've fleshed out 47809c44d001dc3efdba94a5bad62494758d399f in another branch, will open a separate PR for that once this one is merged..